app/models/user.rb

```diff
@@ -35,16 +36,32 @@ def remember
     update_column :remember_digest, User.digest(remember_token)
   end
 
-  def authenticated? remember_token
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def authenticated? attribute, token
+    digest = public_send "#{attribute}_digest"
```

Comment:
public_send khác gì với send em ơi

Reply:
Em xin trả lời: public_send an toàn hơn send khi chỉ cho phép gọi các hàm public của đối tượng, trong khi đó send có thể gọi cả phương thức private, protected của đối tượng, vi phạm tính đóng gói trong lập trình hướng đối tượng và khả năng bảo mật thấp hơn public_send.